src/openassetio-python/tests/CMakeLists.txt

```diff
@@ -19,6 +19,72 @@ add_subdirectory(cmodule)
 add_subdirectory(package)
 
 
+#-----------------------------------------------------------------------
+# Test pip discovers CMake installed package and it has correct version.
+#
+# Doesn't test alpha/beta tags (since they're not part of the CMake
+# version), but does test that the semver version in CMake matches the
+# Python package version.
+
+add_custom_target(
+    openassetio.internal.python-package-pip-discoverable-test
+    COMMAND cmake -E echo --
+    "Ensuring pip can discover CMake installed openassetio with version ${PROJECT_VERSION}"
+    COMMAND
+    $<IF:$<BOOL:${WIN32}>,set,export>
+    PYTHONPATH="${CMAKE_INSTALL_PREFIX}/${OPENASSETIO_PYTHON_SITEDIR}"
+    COMMAND
+    ${OPENASSETIO_PYTHON_EXE} -m pip show openassetio
+)
+openassetio_add_test_target(openassetio.internal.python-package-pip-discoverable-test)
+openassetio_add_test_fixture_dependencies(
+    openassetio.internal.python-package-pip-discoverable-test
+    openassetio.internal.install
+)
+openassetio_add_test_venv_fixture_dependency(openassetio.internal.python-package-pip-discoverable-test)
+set_tests_properties(
+    openassetio.internal.python-package-pip-discoverable-test
+    PROPERTIES
+    PASS_REGULAR_EXPRESSION
+    "Version: ${PROJECT_VERSION}(.|[\r\n])+Location: ${CMAKE_INSTALL_PREFIX}/${OPENASSETIO_PYTHON_SITEDIR}"
+)
+
+
+#-----------------------------------------------------------------------
+# Test that we install a dist-info with no RECORD
+#
+# I.e. that `pip install` will error when attempting to overwrite a
+# CMake installed package.
+
+add_custom_target(
+    openassetio.internal.python-package-no-record-test
+    COMMAND cmake -E echo -- "Check for a files under dist-info"
+    # Abuse md5sum to check existence of METADATA file.
+    COMMAND
+    ${CMAKE_COMMAND} -E md5sum
+    "${CMAKE_INSTALL_PREFIX}/${OPENASSETIO_PYTHON_SITEDIR}/openassetio-${PROJECT_VERSION}*.dist-info/METADATA"
```

Comment:
[question] Whilst were at it, is it worth also checking `INSTALLER` as that ensures we have the friendly message?

Reply:
We could, though this is implicitly tested in CI. Shall I go ahead and test it?  Do we need a test of its contents too?

Reply:
Up to you - if this is more a 'test the structure' test vs a 'test no record' test...

Reply:
Incorporated into pytest tests in dd0e3e6b